File: modules/web-frontend/src/features/school/services/schoolService.ts
import { apiClient } from '../../../lib/axios';

export const schoolService = {
  getAllSchools: async () => {
    const response = await apiClient.get('/schools');
    return response.data;
  },

  searchSchools: async (query: string) => {
    const response = await apiClient.get(`/schools/search?q=${query}`);
    return response.data;
  },

  getSchoolById: async (id: string) => {
    const response = await apiClient.get(`/schools/${id}`);
    return response.data;
  },
};

File: modules/backend-core/src/main/java/ke/elimuconnect/backend/service/SchoolService.java
package ke.elimuconnect.backend.service;

import ke.elimuconnect.domain.School;
import ke.elimuconnect.backend.repository.SchoolRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class SchoolService {
    private final SchoolRepository schoolRepository;

    public List<School> getAllSchools() {
        return schoolRepository.findAll();
    }

    public List<School> searchSchools(String query) {
        return schoolRepository.findByNameContainingIgnoreCase(query);
    }

    public Optional<School> getSchoolById(String id) {
        return schoolRepository.findById(id);
    }
}
